### PR TITLE
Add support for zhimi.humidifier.ca4 (miot)

### PR DESCRIPTION
**Pull request by Toxblh**
Added minimal functionality for work with zhimi.humidifier.ca4 and similar in future humidifiers.

Close one of device from the https://github.com/rytilahti/python-miio/issues/627 issue
 
image

### CODE REVIEW

**Comment by Toxblh** [comment]
@rytilahti Hi, could you help me here, what is wrong with "Labeler / label" check?

**Comment by rytilahti** [comment]
Hey, first of all, thanks for the PR! I will do a code review soon.

Wrt the labeler (which simply adds labels to PRs), looks like it works on repositories it has write access (https://github.community/t/github-actions-are-severely-limited-on-prs/18179), rendering it pretty much useless for this project. I'm going to disable it for the time being, so no need to worry about that.

I'll do a code review soon

File: miio/airhumidifier_miot.py [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return "on" if self.is_on else "off"
+
+    @property
+    def fault(self) -> int:
+        """Fault state."""
+        return self.data["fault"]
```
Comment (rytilahti) [inline miio/airhumidifier_miot.py:84]: Should this be named `error`? Are there any known fault values?

Comment (Toxblh) [reply miio/airhumidifier_miot.py:84]: Nope, in API just wrote about 0 to 15 is possible value. 
Error maybe better yes, agree

File: miio/airhumidifier_miot.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+            return self.data["dry"]
+        return None
+
+    @property
+    def use_time(self) -> int:
+        """How long the device has been active in seconds."""
+        return self.data["use_time"]
+
+    @property
+    def button_pressed(self) -> int:
```
[the remark this answers is not in the document]
Comment (Toxblh) [reply miio/airhumidifier_miot.py:115]: that is consistent with other Xiaomi devices. Like that https://github.com/rytilahti/python-miio/blob/8f16c1b335599e1d02f8217819687843bb687dc8/miio/airpurifier.py#L266
And used in homeassistant also

[the remark this answers is not in the document]
Comment (rytilahti) [reply miio/airhumidifier_miot.py:115]: Fair enough. I think it's worth to start having abstract classes/bases for different device types, to make sure all implementations follow the same (basic) API. The return value of this should then likewise return a str?

[the remark this answers is not in the document]
Comment (Toxblh) [reply miio/airhumidifier_miot.py:115]: In my case, from API I know which Enum will return, I think guys from other parts just don't know that and suggest to use Optional string. In my case better to provide Enum.

File: README.rst [hunk 3]
```diff
@@ -60,11 +60,12 @@ API usage
...
     from miio import Vacuum
-    
+
     vac = Vacuum("<ip address>", "<token>")
     vac.start()
-    
-Each separate device type inherits from `miio.Device` (and in case of miOT devices, `miio.MiotDevice`) which provides common API.
+
+Each separate device type inherits from `miio.Device`
+(and in case of miOT devices, `miio.MiotDevice`) which provides common API.
```
Comment (Toxblh) [inline README.rst:68]:
```Scanning...
Validating...
README.rst:67: D001 Line too long```
prehook pushed to do that

**Pull request merged**